### DISCUSSION TITLE
Importing data from MAUD and how to use the volume function

### DISCUSSION THREAD

**Discussion by cecilgroen** [mtex-toolbox/mtex#979] [Ask Anything]
Hi MTEX community,
I recently started to use MTEX in order to have some more options when plotting my polefigures and to do some calculations from the ODF as the volume-function, but I have some questions and problems I hope some of you can help me solving.

I use: mtex5.4.0 and matlab R2020b

1) I load my data into MTEX by loading all of the polefigures from MAUD, then I calculate the ODF using calcODF(pf, 'noQuadratureWeights'); My problem is that when comparing my texture index from MAUD and MTEX, there is some differences e.g. Tmaud= 4.6386 and Tmtex = 5.1181, sometimes the differences has been even larger. Do I make any mistakes in my loading and calculating procedure? (I will add my script below)

2) For the ODF I want to do some calculations to make a plot of the angular distributions, here I use the volume function: volume(odf,orientation.byMiller([0 0 1],[1 1 0],CS,SS),(theta(j-1))*degree), see line number 73. Here my problems is, when updating to newer versions of mtex (5.6.0) I get the following error:

```
Error using ODF/volume (line 32)
Specimen cymmmetry of odf and center needs to be the same!

Error in testskema (line 73)
    v(j)=volume(odf,orientation.byMiller([0 0 1],[1 1 0],CS,SS),(theta(j-1))*degree);
```
Do you know how to solve this?

My script is the following:
```
1. %% Specify Crystal and Specimen Symmetries
2. 
3. % crystal symmetry
4. CS = crystalSymmetry('6/mmm', [5.9 5.9 23], 'X||a*', 'Y||b', 'Z||c', 'mineral', 'Strontium dodecairon(III) oxide');
5. 
6. 
7. % specimen symmetry
8. SS = specimenSymmetry('orthorhombic');
9. 
10. % plotting convention
11. setMTEXpref('xAxisDirection','north');
12. setMTEXpref('zAxisDirection','outOfPlane');
13. 
14. %% Specify File Names
15. 
16. % path to files
17. pname = 'C:\Users\Me\Documents\data';
18. 
19. % which files to be imported
20. fname = [pname '\SPS200_PF'];
21. 
22. %% Specify Miller Indice
23. 
24. h = { ...
25.   Miller(0,0,0,4,CS),...
26.   Miller(1, 0,-1, 0,CS),...
27.   Miller(1, 0,-1, 1,CS),...
28.   Miller(1, 0,-1, 2,CS),...
29.   Miller(1, 0,-1, 3,CS),...
30.   Miller(0,0,0,6,CS),...
31.   Miller(1, 0,-1, 4,CS),...  
32. %(+ alot more miller indices)
33. };
34. 
35. %% Import the Data
36. 
37. % create a Pole Figure variable containing the data
38. pf = PoleFigure.load(fname,h,CS,SS,'interface','beartex');
39. 
40. %% Import the Data 
41. % create a Pole Figure variable containing the data
42. pf = PoleFigure.load(fname,h,CS,SS,'interface','beartex');
43. 
44. %% odf 
45. % calculate odf from the MAUD polefigures
46. odf = calcODF(pf, 'noQuadratureWeights');
47. 
48. %calculate texture index
49. tex_index=textureindex(odf);
50. 
51. %% recalculated polefigures from odf
52. figure(1);
53. plotPDF(odf,Miller({0,0,0,8},CS));
54. colormap jet
55. CLim(gcm,[0,15])
56. 
57. 
58. %% angular distribution function
59. %Normalization to volume - volume of hemisphere
60. if 1
61. v(1)=0;
62. vol_hemis=2/3*pi;
63. i=[1:25];
64. 
65. % hemisphere is divided into equal volume pieces
66. volfrac=vol_hemis/max(i);
67. theta=acos(1-i*volfrac/vol_hemis)*180/pi;
68. 
69. for j=2:length(theta)+1;
70.     j;
71.     % The angular interval between these points
72.     % Based on the volume of a cone of a sphere Vcone=2/3piR^3(1-cos(theta))
73.     v(j)=volume(odf,orientation.byMiller([0 0 1],[1 1 0],CS,SS),(theta(j-1))*degree);
74.     v_int(j-1)=v(j)-v(j-1);
75. end
76. 
77. for i=2:2:2*length(theta)
78.     theta_new(i)=theta(i/2);
79.     theta_new(i+1)=theta(i/2);
80.     v_int_new(i-1)=v_int(i/2);
81.     v_int_new(i)=v_int(i/2);
82. end
83. 
84. theta_new(end)=[];
85. outtex=[theta_new;v_int_new]';
86. 
87. figure
88. plot(theta_new,v_int_new,'-')
89. ylim([0,0.30]);
90. end
91. 
92. xlabel('angle')
93. ylabel('[001] ratio / volume')

```

**Comment by kilir** [comment]
Hi,
1) the textureindex strongly depends on the way you compute the ODF, e.g. your choice of a kernel width / bandwidth. How does MAUD represent the ODF?

2) The reason for  the error is in the error message. Instead of 

```
volume(odf,orientation.byMiller([0 0 1],[1 1 0],CS,SS),(theta(j-1))*degree);
```
you could do
```
volume(odf,orientation.byMiller([0 0 1],[1 1 0],CS,odf.SS),(theta(j-1))*degree);
```

Alternatively, I guess you odf has '1' as specimen symmetry, so you can simply omit the SS.


Cheers,
Rüdiger

**Reply by cecilgroen** [reply]
For MAUD I use the EWIMV model to calculate the ODF while refining my data sets. Do I need to change the model in MTEX in order to copy the the model from MAUD better? Or are the differences something I just have to accept? I guess I am using the standard method in MTEX, since I haven't specified it. 
(I can also choose to export a ODF file from MAUD, but I haven't been successful, when I have tried to import it into MTEX.)

It helped changing SS to odf.SS. I have orthorhombic as specimen symmetry, so it was necessary, thank you!!

Cheers, 
Cecilie
